src/project.coffee

```diff
@@ -271,9 +274,51 @@ class Project extends Model
     @rootDirectories.some (dir) -> dir.contains(pathToCheck)
 
   ###
+  Section: Environment
+  ###
+
+  # Public: Retrieves a normalized copy of the environment.
+  #
+  # On OS X, the `PATH` can be different depending on whether Atom is launched
+  # from the Dock, Spotlight or the terminal. This detects how Atom was started
+  # and corrects the `PATH` environment variable before returning a copy of the
+  # environment.
+  getEnv: ->
+    unless @env?
+      @env = _.clone(process.env)
+      if process.platform is "darwin" and not process.env.TERM?
+        @env.PATH = @getShellPath()
+
+    _.clone(@env)
+
+  ###
   Section: Private
   ###
 
+  getShellPath: ->
+    shellEnvText = @getShellEnv()
+    env = {}
+
+    for line in shellEnvText.split(os.EOL)
+      if line.includes("=")
+        components = line.split("=")
+        if components.length is 2
+          env[components[0]] = components[1]
+        else
+          k = components.shift()
+          v = components.join("=")
+          env[k] = v
+
+    env.PATH
+
+  getShellEnv: ->
+    shell = process.env.SHELL ? "/bin/bash"
+    results = child_process.spawnSync shell, ["--login"], input: "env", encoding: "utf8"
```

Comment:
I just tried it and it doesn't give accurate terminal results for `$PATH` me on my mac
```
# From terminal
/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/steel/n/bin:/usr/games/bin
```
```
# From executing this
/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin
```